### PR TITLE
closedts: make settings TenantReadOnly and public

### PR DESCRIPTION
**Pull request by erikgrinaker**
It doesn't make sense for these to be `TenantWritable`, since the side transport runs below KV. Furthermore, these settings are referenced throughout our documentation, so make them public.

These should really be set only for the system tenant, and secondary tenants could simply read the system tenant's setting. This functionality runs in the host cluster below KV and it doesn't make any sense to set individual settings for tenants here. Unfortunately, this isn't currently possible with the existing settings classes, there is no way for secondary tenants to access the host's settings.

Touches #108677.

Epic: none
Release note (ops change): The following closed timestamp side-transport settings can no longer be set from secondary tenants (they did not have an effect in secondary tenants): kv.closed_timestamp.target_duration, kv.closed_timestamp.side_transport_interval, and kv.closed_timestamp.lead_for_global_reads_override.

### CODE REVIEW

**Comment by erikgrinaker** [comment]
We can't easily do this, because stuff like `follower_read_timestamp()` currently relies on this. It really shouldn't though.

**Comment by erikgrinaker** [comment]
Downgrading to draft while I work through the test failures.

**Comment by erikgrinaker** [comment]
@aliher1911  This should be good to go. It's a prereq for #108667.

cc @nvanbenschoten in case you have opinions on closed timestamp settings.

File: pkg/kv/kvserver/closedts/setting.go [hunk 1]
```diff
@@ -16,33 +16,41 @@ import (
...
 
+// NB: These settings are TenantReadOnly because they need to be read by e.g.
+// rangefeed clients and follower_read_timestamp(). However, they really need
+// to see the host's setting, not the tenant's setting. See:
+// https://github.com/cockroachdb/cockroach/issues/108677
+
 // TargetDuration is the follower reads closed timestamp update target duration.
 var TargetDuration = settings.RegisterDurationSetting(
-	settings.TenantWritable,
+	settings.TenantReadOnly,
```
Comment (stevendanna) [inline pkg/kv/kvserver/closedts/setting.go:26]: Do we need to call this out as a breaking change? If users are erroneously setting these in their tenants in automation, they will now get an error.

Comment (erikgrinaker) [reply pkg/kv/kvserver/closedts/setting.go:26]: It only affects serverless users. Not sure how we communicate breaking changes there, but release notes don't seem like the best channel. Will find out.

**Review by nvb — APPROVED** [review]
> cc [@nvanbenschoten](https://github.com/nvanbenschoten) in case you have opinions on closed timestamp settings.

The change here :lgtm:. It's unfortunate that we need to manually fanout the cluster settings when we want to change them on the host, but that isn't changing in this PR and it's still better to not let the setting have write access to the setting.

> Reviewed 14 of 14 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/108678#-:-NcNWfj6CiuDQ5tOOFQa:b-25sxh7)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @aliher1911 and @stevendanna)

**Comment by erikgrinaker** [comment]
TFTR!

bors r+

File: pkg/kv/kvserver/closedts/setting.go (hunk 1 above)
Comment (erikgrinaker) [reply pkg/kv/kvserver/closedts/setting.go:26]: Added a release note.

**Pull request merged**